harness/determined/util.py

```diff
@@ -246,6 +248,37 @@ def jsonable(obj: Any) -> Any:
     return json.dumps(jsonable(obj), indent=indent, sort_keys=sort_keys)
 
 
+def _make_shutil_ignore(root_path: pathlib.Path) -> Callable:
+    ignore = list(common_constants.DEFAULT_DETIGNORE)
+    ignore_path = root_path / ".detignore"
+    if ignore_path.is_file():
+        with ignore_path.open("r") as detignore_file:
+            ignore.extend(detignore_file)
+
+    import pathspec
+
+    ignore_spec = pathspec.PathSpec.from_lines(pathspec.patterns.GitWildMatchPattern, ignore)
+
+    def _ignore(path: str, names: List[str]) -> Set[str]:
+        ignored_names = []  # type: List[str]
+        for name in names:
+            if name == ".detignore":
+                ignored_names.append(name)
+                continue
+
+            file_path = pathlib.Path(path) / name
+            file_rel_path = file_path.relative_to(root_path)
+
+            if ignore_spec.match_file(str(file_rel_path)) or ignore_spec.match_file(
+                str(file_rel_path) + "/"
```

Comment:
if I understand correctly, you're testing with the `/` in the name for regular files as well, which is different than the other implementation.

Technically you're also ignoring .detignore directories too, but that feels like not a huge deal.  But probably if you fix the first issue fixing this one will be trivial.